packages/vite/src/node/utils.ts

```diff
@@ -162,6 +162,17 @@ export const deepImportRE = /^([^@][^/]*)\/|^(@[^/]+\/[^/]+)\//
 // TODO: use import()
 const _require = createRequire(import.meta.url)
 
+export function resolveDependencyVersion(
+  dep: string,
+  pkgRelativePath = '../../package.json',
+): string {
+  const pkgPath = path.resolve(_require.resolve(dep), pkgRelativePath)
```

Comment:
What if rollup or esbuild changes the file structure? 👀 

Shouldn't `require.resolve(${dep}/package.json)` work better?

Reply:
We get `Package subpath './package.json' is not defined by "exports"` if we try to do that

Reply:
I'm also worried about a structure change but I would imagine they won't do outside of a major. We were already doing it this way until #10964, see https://github.com/vitejs/vite/pull/10964/files

Reply:
I see. I thought it's a standard practice to put a package json in exports field. I am fine with the change then 👍

Reply:
If we want to do this right, we can also use the `resolvePackageData` util. Though the `loadPackageData` flow might slow it a bit.